improve autotool

LordMZTE opened the pull request:
autotool will now always select the correct tool, also for special cases like bamboo by using the same method as the game for calculating the break speed

LordMZTE committed:
improve autotool
now works with bamboo too

LordMZTE commented:
closes #306

zeroeightysix commented on src/main/kotlin/me/zeroeightsix/kami/feature/module/AutoTool.kt:
```diff
@@ -44,35 +43,33 @@ object AutoTool : Module() {
-                }
+            // this does not send the selected slot to the server
+            // we do this in order to make calcBlockBreakingDelta use the correct slot
+            equip(i, false)
```
Instead of placing a comment, just explicitly specify `updateServer`: `equip(i, updateServer = false)`. It is more concise.

zeroeightysix commented on src/main/kotlin/me/zeroeightsix/kami/feature/module/AutoTool.kt:
```diff
@@ -44,35 +43,33 @@ object AutoTool : Module() {
+            // we do this in order to make calcBlockBreakingDelta use the correct slot
+            equip(i, false)
+            // this function is not actually deprecated. mojang just doesn't know what deprecated means
+            val delta = blockState.block.calcBlockBreakingDelta(blockState, mc.player, mc.world, blockPos)
```
If it's marked deprecated, I wouldn't be surprised if it disappears in a version to come. However, for now, instead of taking a jab at mojang, let's just suppress the warning:
```suggestion
            @Suppress("DEPRECATION")
            val delta = blockState.block.calcBlockBreakingDelta(blockState, mc.player, mc.world, blockPos)
```

zeroeightysix commented on src/main/kotlin/me/zeroeightsix/kami/feature/module/AutoTool.kt:
```diff
@@ -107,12 +106,15 @@ object AutoTool : Module() {
 
     enum class PreferredWeapon(val item: Class<out Item>, val damage: (ItemStack) -> Float?) {
-        SWORD(SwordItem::class.java, { (it.item as? SwordItem)?.attackDamage }), AXE(AxeItem::class.java, { (it.item as? AxeItem)?.attackDamage })
+        SWORD(SwordItem::class.java, { (it.item as? SwordItem)?.attackDamage }), AXE(
+            AxeItem::class.java,
+            { (it.item as? AxeItem)?.attackDamage })
```
```suggestion
        SWORD(SwordItem::class.java, { (it.item as? SwordItem)?.attackDamage }),
        AXE(AxeItem::class.java, { (it.item as? AxeItem)?.attackDamage })
```

LordMZTE committed:
implement requested changes to autotool
supress deprecation
mob mojang a little less
add verbosity to updating slot without notifying server
fix format

zeroeightysix reviewed (approved):
Thanks!
